- Manifest changes to include new RRS (Rack Resiliency Service) helm chart

ravikanth-nalla-hpe opened the pull request:
## Summary and Scope
Manifest changes to include new RRS (Rack Resiliency Service) helm chart

## Issues and Related PRs
https://jira-pro.it.hpe.com:8443/browse/CASM-5603
https://jira-pro.it.hpe.com:8443/browse/CASM-5602

Implementation of the Rack Resiliency Service (RRS)

## Testing

_List the environments in which these changes were tested._

### Tested on:

Internal development system
Local development environment
Surtur
Odin

### Test description:
Pod is launched successfully, and all containers are running and healthy
Init container is modifying the dynamic CM with required information and exiting
RMS container is running as expected
- Performing infinite loop monitoring critical services, ceph status and updating k8s/CEPH zone placements
- receiving requests from HMNFD and perform monitoring
API backend is serving CLI commands which are returning expected results

## Risks and Mitigations

None

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [] License file intact
- [x] Target branch correct
- [x] Testing is appropriate and complete, if applicable
- [] Release Notes in [docs-csm](https://github.com/Cray-HPE/docs-csm) updated, if applicable

ravikanth-nalla-hpe committed:
- Manifest changes to include RRS (Rack Resiliency Service) helm chart

dborman-hpe commented:
Question: We agreed that RR would be disabled by default, and only enabled once the customer has confirmed that their hardware configuration is properly set up for RR.  Your testing description sounds like once this is added to the manifests, it'll be running by default (the `updating k8s/CEPH zone placements` in the testing description).

Can you confirm that once this manifest PR merges that RR will only be running on systems that have explicitly enabled it?

In other words, I'm looking for assurance that adding this to the manifest won't break systems that haven't been properly configured for RR.

mtupitsyn commented:
FYI - validation check is currently failing:

    [2025-06-26T19:25:50.645Z] time="2025-06-26T19:25:50Z" level=fatal msg="Error parsing image name \"docker://artifactory.algol60.net/csm-docker/stable/cray-rrs:1.0.0\": reading manifest 1.0.0 in artifactory.algol60.net/csm-docker/stable/cray-rrs: manifest unknown: The named manifest is not known to the registry."

ravikanth-nalla-hpe commented:
> cker://artifactory.algol60.net/csm-docker/stable/cray-r

Yes, I saw it, we are fixing it. Will bump-up the chart version. Thanks.

mtupitsyn commented:
Please also propose the change via JIRA Task or Subtask. All 3 tickets mentioned in this PR are Epics, we do not accept Epics in the release.

ravikanth-nalla-hpe committed:
- Updated RRS chart version

ravikanth-nalla-hpe commented:
> Question: We agreed that RR would be disabled by default, and only enabled once the customer has confirmed that their hardware configuration is properly set up for RR. Your testing description sounds like once this is added to the manifests, it'll be running by default (the `updating k8s/CEPH zone placements` in the testing description).
> 
> Can you confirm that once this manifest PR merges that RR will only be running on systems that have explicitly enabled it?
> 
> In other words, I'm looking for assurance that adding this to the manifest won't break systems that haven't been properly configured for RR.

It will be handled part of post hook https://jira-pro.it.hpe.com:8443/browse/CASM-5499 where it will check for RR enablement also zones creation and then only deploy the RRS chart. For now, we want to get into platform.yaml file to get this deployed (where we enable RR beforehand and verify the feature).

ravikanth-nalla-hpe commented:
> Please also propose the change via JIRA Task or Subtask. All 3 tickets mentioned in this PR are Epics, we do not accept Epics in the release.

Updated 

> Please also propose the change via JIRA Task or Subtask. All 3 tickets mentioned in this PR are Epics, we do not accept Epics in the release.

Updated the PR with respective JIRAs.
